internal/services/expense_service.go

```diff
@@ -0,0 +1,111 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/Yokithaii/CLI_Expense_Tracker/internal/dataaccess"
+	"github.com/Yokithaii/CLI_Expense_Tracker/internal/entities"
+)
+
+type ExpenseService struct {
+	expenseDAO dataaccess.ExpenseDAO // Для работы с БД сервису нужен будет data access object для наших expense-ов
+}
+
+func NewConnectionToDB() (*ExpenseService, error) {
+	expService, err := dataaccess.NewConnectionToDB()
+	if err != nil {
+		return nil, err
+	}
+	return &ExpenseService{*expService}, nil
+}
+
+func (s *ExpenseService) Close(context context.Context) {
+	s.expenseDAO.CloseConnect(context)
+}
+
+// Validates provided data.
+func validateExpenseData(amount int, description string) error {
+	if amount < 0 || strconv.Itoa(amount)[0] != '0' {
+		return errors.New("bad amount promoted")
+	}
+	runes := []rune(description)
+	if len(runes) > 30 {
+		return errors.New("your description is too long")
+	}
+	return nil
+}
+
+// Checks if expense exists.
+func (s *ExpenseService) validateExpenseExistence(id int) error {
+	exists, err := s.expenseDAO.ExpenseExists(id)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("no expense with given id")
+	}
+	return nil
+}
+
+// Adds a promoted expense (need to validate).
+func (s *ExpenseService) CreateExpense(expense entities.Expense) error {
+	err := validateExpenseData(expense.Amount, expense.Desc)
+	if err != nil {
+		return err
+	}
+	return s.expenseDAO.CreateExpense(expense)
+}
+
+// Deletes an expense with promoted id.
+func (s *ExpenseService) DeleteExpense(id int) error {
+	err := s.validateExpenseExistence(id)
+	if err != nil {
+		return err
+	}
+
+	return s.expenseDAO.DeleteExpense(id)
+}
+
+// Replaces expense with promoted id with new given expense.
+func (s *ExpenseService) UpdateExpense(id int, expense entities.Expense) error {
+	err := validateExpenseData(expense.Amount, expense.Desc)
+	if err != nil {
+		return err
+	}
+
+	exists, err := s.expenseDAO.ExpenseExists(id)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("no expense with given id")
+	}
+
+	return s.UpdateExpense(id, expense)
+}
+
+// Returns an array of all expenses available.
+func (s *ExpenseService) GetAllExpenses() ([]entities.Expense, error) {
+	expenses, err := s.expenseDAO.GetAllExpensesFromDb()
+	if err != nil {
+		return nil, err
+	}
+	return expenses, nil
+}
+
+// Returns an amount of all expenses.
+func (s *ExpenseService) Summary() (int, error) {
+	sum, err := s.expenseDAO.Summary()
+	if err != nil {
+		return -1, err
```

Comment:
```suggestion
		return 0, err
```
Обычно в случае ошибки возвращают дефолтное значение. Для числа это - 0.

Это не догма, от нее можно отклоняться, но в данном случае нет повода не следовать общей практике.